realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -281,6 +283,35 @@ public void refresh() {
     }
 
     /**
+     * Blocks the current thread util transactions are committed by other threads to this Realm or
+     * {@link #stopWaitForChange()} is called from other threads.
+     *
+     * @return {@code true} if transactions are committed to this Realm, {@code false} if
+     * {@link #stopWaitForChange()} gets called.
+     * @throws IllegalStateException if attempting to wait within a transaction.
+     */
+    public boolean waitForChange() {
+        checkIfValid();
+        if (isInTransaction()) {
+            throw new IllegalStateException("Cannot wait for changes inside of a transaction.");
+        }
+        sharedGroupManager.setWaitForChangeEnabled(true);
+        boolean hasChanged = sharedGroupManager.waitForChange();
+        if (hasChanged) {
+            sharedGroupManager.advanceRead();
+        }
+        return hasChanged;
+    }
+
+    /**
+     * Makes {@link #waitForChange()} return {@code false} immediately. This method is supposed to
```

Comment:
Maybe first sentence should be: Cancel all outstanding waitForChange() calls, forcing them to return false.